lib/internal/process/execution.js

```diff
@@ -64,6 +64,19 @@ function evalScript(name, body, breakFirstLine, print) {
   const module = new CJSModule(name);
   module.filename = path.join(cwd, name);
   module.paths = CJSModule._nodeModulePaths(cwd);
+
+  let fileURL;
+  module.importModuleDynamically = async (specifier) => {
```

Comment:
Is it worth trying to consolidate the logic for `importModuleDynamically`, perhaps a `TODO(@coreyfarrell) investigate whether this logic could be merged`.

Either way, I think a comment above both instances of `importModuleDynamically` that explains what context they will run in would be helpful for future generations 😆

Reply:
Is there a reason we can't just inline this implementation into the `runInThisContext` call directly? Why does it need to go through the `module` object specifically?

Reply:
I went ahead and pushed a commit to try and inline more of the logic in the spirit of @bcoe's comment. I hope that works for both of you here.

In terms of how we handle `importModuleDynamically`, it seems to make sense that it should be possible to have a single default implementation for all modules created by the _primary loader_, to avoid the need for a custom callback map function entry and call for each CJS and ESM module. But that can be considered as a future refactoring I think.

Reply:
@guybedford thanks for updating this for me, sorry I got busy/distracted with other stuff and dropped the ball on this.  I'm unsure why I used the `module` object instead of `global`.  `global` makes sense since we're already using it to pass the other internal into `runInThisContext`.

Regarding `importModuleDynamically` I think we need two separate implementations but the can be unified.  One for when it is run by an ES module where the second argument has a `url` property, the other for when it is run by other means where the referred URL needs to be provided from the outside.

The alternative might be if `import.meta.url` were made available to CJS modules with `import.meta.url === url.pathToFileURL(__filename)`, then I suspect we could have a universal `importModuleDynamically` function or maybe it could be optional to even register it.

In either case I feel that de-duplication of `importModuleDynamically` is out of scope for this bug-fix.